nixos/tests/non-switchable-system.nix

```diff
@@ -0,0 +1,15 @@
+{ lib, ... }:
+
+{
+  name = "non-switchable-system";
+
+  meta.maintainers = with lib.maintainers; [ nikstur ];
+
+  nodes.machine = {
+    system.switch.enable = false;
+  };
+
+  testScript = ''
+    machine.fail("test -e /run/current-system/bin/switch-to-configuration")
+  '';
+}
```

Comment:
Nothing else to be tested?
Maybe test that your profile doesn't have `perl`? Or isn't it quite there yet?

[system.forbiddenDependenciesRegex](https://search.nixos.org/options?channel=23.05&show=system.forbiddenDependenciesRegex&from=0&size=50&sort=relevance&type=packages&query=forbidden)

Reply:
Not quite there yet, but I have some more work in the pipeline ;)